_tools/generate_release_cycle.py

```diff
@@ -28,21 +28,52 @@ def parse_date(date_str: str) -> dt.date:
 class Versions:
     """For converting JSON to CSV and SVG."""
 
-    def __init__(self) -> None:
+    def __init__(self, limit_to_active=False, special_py27=False) -> None:
         with open("include/release-cycle.json", encoding="UTF-8") as in_file:
             self.versions = json.load(in_file)
 
         # Generate a few additional fields
         for key, version in self.versions.items():
             version["key"] = key
-            version["first_release_date"] = parse_date(version["first_release"])
+            version["first_release_date"] = r1 = parse_date(version["first_release"])
+            version["start_security_date"] = r1 + dt.timedelta(days=2 * 365)
             version["end_of_life_date"] = parse_date(version["end_of_life"])
+
+        self.cutoff = min(ver["first_release_date"] for ver in self.versions.values())
+
+        if limit_to_active:
+            self.cutoff = min(
+                version["first_release_date"]
+                for version in self.versions.values()
+                if version["status"] != 'end-of-life'
+            )
+            self.versions = {
+                key: version
+                for key, version in self.versions.items()
+                if version["end_of_life_date"] >= self.cutoff
+                or (special_py27 and key == '2.7')
+            }
+            if special_py27:
+                self.cutoff = min(self.cutoff, dt.date(2019, 8, 1))
+            self.id_key = 'active'
+        else:
+            self.id_key = 'all'
+
         self.sorted_versions = sorted(
             self.versions.values(),
             key=lambda v: [int(i) for i in v["key"].split(".")],
             reverse=True,
         )
 
+        # Set the row (y-coordinate) for the chart, to allow a gap between 2.7
```

Comment:
We have "X coordinate" elsewhere
```suggestion
        # Set the row (Y coordinate) for the chart, to allow a gap between 2.7
```